test/src/multi_thread_matrix_test.cpp

```diff
@@ -469,5 +469,104 @@ TEST_F(TestMatrixMultiThread, assignmentFromVector) {
     ASSERT_EQ(a.dataPtr(), nullptr);
     ASSERT_EQ(b.dataPtr(), nullptr);
 }
+
+TEST_F(TestMatrixMultiThread, transpose) {
```

Comment:
This file is for testing the methods in `matrix.h`, so the first two unit tests should not be here. They should be moved to `test/src/multi_thread_matrix_calculation_test.cpp`.